projects/ion/src/lib/pagination/utils/ellipsis.component.ts

```diff
@@ -0,0 +1,110 @@
+import { IonPaginationComponent } from '../pagination.component';
+
+export interface Limit {
+  leftLimit: number;
+  showLeftEllipsis: boolean;
+  firstPages: boolean;
+}
+export class IonPaginationEllipsisComponent {
+  constructor(private paginationComponent: IonPaginationComponent) {}
+
+  createEllipsis(index: number, qtdOfPages: number): void {
+    if (index === 1) {
+      this.createBothEllipsis(-1);
+    } else if (index === qtdOfPages - 1) {
+      this.createBothEllipsis(0);
+    }
+    this.paginationComponent.pages.push({
+      selected: false,
+      page_number: index + 1,
+    });
+  }
+
+  createBothEllipsis(number: number): number {
+    return this.paginationComponent.pages.push({
+      selected: false,
+      page_number: number,
+    });
+  }
+
+  skipEllipsis(pageNumber: number): number {
+    if (pageNumber === 1) return pageNumber - 1;
+    else if (pageNumber === this.paginationComponent.pages.length - 2)
```

Comment:
Nesse caso o else não é necessário, pois, se o primeiro if for false, ele verificar o segundo de qualquer forma. Se o primeiro for true, como há um return, ele não ira verificar o restante do código da função.